src/utils/dateUtils.spec.ts

```diff
@@ -0,0 +1,73 @@
+import { ZetkinEvent } from '../types/zetkin';
+import { getFirstAndLastEvent, removeOffset } from './dateUtils';
+
+const firstEvent = {
+    activity: {
+        title: 'Active activity',
+    },
+    campaign: {
+        id: 1,
+        title: 'Testcampaign Title',
+    },
+    contact: null,
+    end_time: '2022-06-16T09:00:00+00:00',
+    id: 5,
+    info_text: 'Info text is informational',
+    location: {
+        id: 1,
+        lat: 51.192702,
+        lng: 12.284873,
+        title: 'Dorfplatz',
+    },
+    num_participants_available: 3,
+    num_participants_required: 2,
+    organization: {
+        id: 1,
+        title: 'My Organization',
+    },
+    start_time: '2022-06-16T07:00:00+00:00',
+    title: undefined,
+    url: undefined,
+};
+
+const secondEvent = {
+    ...firstEvent,
+    end_time: '2022-06-17T09:00:00+00:00',
+    start_time: '2022-06-17T07:00:00+00:00',
+};
+
+const thirdEvent = {
+    ...firstEvent,
+    end_time: '2022-06-18T09:00:00+00:00',
+    start_time: '2022-06-18T07:00:00+00:00',
+};
+
+const fourthEvent = {
+    ...firstEvent,
+    end_time: '2022-06-19T09:00:00+00:00',
+    start_time: '2022-06-19T07:00:00+00:00',
+};
```

Comment:
A third approach could be to create a suite of testing utility functions, starting with just one for this specific use case called `createMockEvent(specifics)` that merges `specifics` with a dummy.

That's one of several strategies mentioned in a talk I saw once, and I like it because it hides away the unrelated stuff, but makes it very clear to the reader of the test what's relevant for the test (whatever is in `specifics`):

```tsx
const myEvent = createMockEvent({
  end_time: '…',
  start_time: '…',
});
```

I also think that these should be created inside the test that uses them, not globally like here. That makes it even easier to read, because it makes it clear what properties are relevant to the test in which the mock is created.

The presentation might have been https://www.youtube.com/watch?v=MU7K_V6rFjM or perhaps https://vimeo.com/53276460. I can't remember which, but both are good for someone who's new to TDD, e.g. @ziggabyte? 😊

Reply:
Yet another approach is to use something like [`ts-mockito`](https://github.com/NagRock/ts-mockito) that I've used with great success on the back-end.